SonarQube.TeamBuild.PreProcessor/SonarWebService.cs

```diff
@@ -182,6 +186,25 @@ public bool TryDownloadEmbeddedFile(string pluginKey, string embeddedFileName, s
 
         #endregion
 
+        #region Private methods
+
+        /// <summary>
+        /// Concatenates project key and branch into one string.
+        /// </summary>
+        /// <param name="projectKey">Unique project key</param>
+        /// <param name="projectBranch">Specified branch of the project. Null if no branch to be specified.</param>
+        /// <returns>A correctly formatted branch-specific identifier (if appropriate) for a given project.</returns>
+        private static string GetProjectIdentifier(string projectKey, string projectBranch = null)
```

Comment:
There is similar logic to compute the projectId in GetSonarDashboadUrl method from the SummaryReportBuilder class. These should be consolidated.